packages/desktop-client/src/components/reports/Charts.js

```diff
@@ -0,0 +1,178 @@
+}) {
+  const hoverStyle = {
+    color: colors.b4,
+    cursor: 'pointer',
+    backgroundColor: 'inherit',
+  };
+  const linkStyle = [
+    header && styles.largeText,
+    {
+      color: colors.n7,
+      fontWeight: bold ? fontWeight : null,
+    },
+    { ':hover': hoverStyle },
+  ];
+
+  const content = (
+    <View
+      style={{
+        flexDirection: 'row',
+        alignItems: 'center',
+        height: 16,
+        color: reportPage === id ? colors.n3 : 'inherit',
+      }}
+    >
+      {Icon && (
+        <Icon id={id} width={12} height={12} style={{ color: 'inherit' }} />
+      )}
+      <Block id={id} style={{ marginLeft: Icon ? 3 : 0, color: 'inherit' }}>
+        {title}
+      </Block>
+    </View>
+  );
+
+  return (
+    <View style={[{ flexShrink: 0 }, style]}>
+      <Button
+        id={title + 'Button'}
+        style={linkStyle}
+        onClick={() => handleClick(id)}
+        bare
+      >
+        {content}
+      </Button>
+    </View>
+  );
+}
+
+export function ChartExtraColumn({
+  start,
+  end,
+  totalExpenses,
+  totalIncome,
+  selectList,
+}) {
+  let amt =
+    selectList === 'Expense'
+      ? totalExpenses
+      : selectList === 'Income'
+      ? totalIncome
+      : totalExpenses + totalIncome;
+  let net = totalExpenses > totalIncome ? 'EXPENSE' : 'INCOME';
+  return (
+    <View
+      style={{
+        padding: 15,
+        overflow: 'auto',
+        flexDirection: 'column',
+        flex: 1,
+        marginTop: 15,
+        marginRight: 30,
+        marginLeft: 10,
+      }}
+    >
+      <View
+        style={{
+          backgroundColor: colors.n11,
+          height: 100,
+          alignItems: 'center',
+          justifyContent: 'center',
+        }}
+      >
+        <Text
+          style={[
+            styles.largeText,
+            {
+              alignItems: 'center',
+              marginBottom: 2,
+              fontWeight: 600,
+            },
+          ]}
+        >
+          {monthUtils.format(start, 'MMM yyyy')} -{' '}
+          {monthUtils.format(end, 'MMM yyyy')}
+        </Text>
+      </View>
+      <View
+        style={{
+          backgroundColor: colors.n11,
+          height: 100,
+          justifyContent: 'center',
+          alignItems: 'center',
+          marginTop: 10,
+        }}
+      >
+        <Text
+          style={[
+            styles.mediumText,
+            {
+              alignItems: 'center',
+              marginBottom: 2,
+              fontWeight: 400,
+            },
+          ]}
+        >
+          {selectList === 'Expense'
+            ? 'TOTAL SPENDING'
+            : selectList === 'Income'
+            ? 'TOTAL INCOME'
+            : 'NET ' + net}
+        </Text>
+        <Text
+          style={[
+            styles.veryLargeText,
+            {
+              alignItems: 'center',
+              marginBottom: 2,
+              fontWeight: 800,
+            },
+          ]}
+        >
+          {integerToCurrency(amt)}
```

Comment:
I’m not sure why, but when I switch to the I&E tab it briefly displays a negative value here before correcting it to positive. (The number doesn’t change, just the sign). Totally fine to file a bug to fix later since it’s not a big deal but if you can track it down that would be nice!

Reply:
I was trying to create a positive value when it's displayed alone. But I think it's better to just let it be negative to avoid confusion. It is an expense after all.